packages/synapse-react-client/src/components/TableColumnSchemaEditor/ColumnModelForm.tsx

```diff
@@ -0,0 +1,392 @@
+import { HIERARCHY_VERTICAL_LINE_COMPONENT } from './TableColumnSchemaForm'
+import { InfoTwoTone } from '@mui/icons-material'
+
+type ColumnOrSubcolumnFormProps = {
+  entityType: EntityType
+  columnModelIndex: number
+  jsonSubColumnIndex?: number
+}
+const jsonSubColumnFieldSx: SxProps = {
+  height: '28px',
+  fontSize: '12px',
+}
+const topLevelColumnModelFieldSx: SxProps = {
+  height: '38px',
+  fontSize: '14px',
+}
+
+export default function ColumnModelForm(props: ColumnOrSubcolumnFormProps) {
+  const { columnModelIndex, jsonSubColumnIndex, entityType } = props
+  const isJsonSubColumn = jsonSubColumnIndex != undefined
+  const dispatch = useSetAtom(tableColumnSchemaFormDataAtom)
+  const isView = (VIEW_CONCRETE_TYPE_VALUES as readonly string[]).includes(
+    convertToConcreteEntityType(entityType),
+  )
+
+  const [isShowingRestrictedValuesModal, setIsShowingRestrictedValuesModal] =
+    useState(false)
+
+  const columnModelAtom = useMemo(
+    () =>
+      selectAtom(
+        tableColumnSchemaFormDataAtom,
+        v =>
+          isJsonSubColumn
+            ? v[columnModelIndex].jsonSubColumns![jsonSubColumnIndex]
+            : v[columnModelIndex],
+        isEqual,
+      ),
+    [columnModelIndex, isJsonSubColumn, jsonSubColumnIndex],
+  )
+
+  const columnModel = useAtomValue(columnModelAtom)
+
+  const allowedColumnTypes = useMemo(
+    () => getAllowedColumnTypes(isView, isJsonSubColumn),
+    [isView, isJsonSubColumn],
+  )
+
+  const allowedFacetTypes = useMemo(
+    () => configureFacetsForType(columnModel.columnType, isJsonSubColumn),
+    [columnModel.columnType, isJsonSubColumn],
+  )
+  const fieldSx: SxProps = useMemo(
+    () => (isJsonSubColumn ? jsonSubColumnFieldSx : topLevelColumnModelFieldSx),
+    [isJsonSubColumn],
+  )
+
+  return (
+    <>
+      {isJsonSubColumn && (
+        <Box sx={{ gridColumn: '1 / span 1' }}>
+          {HIERARCHY_VERTICAL_LINE_COMPONENT}
+        </Box>
+      )}
+      <Box
+        display={'flex'}
+        alignItems={'center'}
+        sx={{
+          gridColumn: isJsonSubColumn ? '2 / span 1' : ' 1 / span 1',
+        }}
+      >
+        <Checkbox
+          label={'Select'}
+          hideLabel
+          checked={columnModel.isSelected}
+          onChange={() => {
+            dispatch({
+              type: 'toggleSelect',
+              columnModelIndex,
+              jsonSubColumnModelIndex: jsonSubColumnIndex,
+            })
+          }}
+        />
+      </Box>
+      <Box
+        sx={{
+          gridColumn: isJsonSubColumn
+            ? /* If this is a JSON Subcolumn, we reduce the width of this grid column to create space to render the visual hierarchical line */
+              '3 / span 1'
+            : /* Otherwise, span across both grid columns  */
+              '2 / span 2',
+        }}
+      >
+        <TextField
+          value={columnModel.name}
+          placeholder={isJsonSubColumn ? 'Facet name' : 'Column name'}
```

Comment:
Regarding "Facet name".  If it is a json sub column, it's a facet?

Reply:
Correct. Facet stats/visualization are the only reasons you would want to define a jsonSubColumn.